variable/cumulative.cpp

```diff
@@ -4,19 +4,18 @@
 /// @author Simon Heybrock
 #include "scipp/variable/cumulative.h"
 #include "scipp/core/element/cumulative.h"
+#include "scipp/variable/creation.h"
 #include "scipp/variable/transform.h"
-#include "scipp/variable/util.h"
 
 using namespace scipp;
 
 namespace scipp::variable {
 
 Variable cumsum(const VariableConstView &var, const Dim dim,
                 const CumSumMode mode) {
+  Variable cumulative = make_accumulant(var, dim, FillValue::ZeroNotBool);
   if (var.dims()[dim] == 0)
```

Comment:
Is this check still necessary? Does `accumulate_in_place` misbehave otherwise?

Reply:
I did this mainly for consistency between the algorithms.
But now looking at it again, I think this changes the logic a bit. Before in the `var.dims()[dim] == 0` case, `cumsum` returned just a copy of `var` which still includes the empty dimension. But with this change, it removes that dim. I think the original approach is better because `cumsum` keeps the summed dim in other cases.

Reply:
I did this mainly for consistency between the algorithms.
But now looking at it again, I think this changes the logic a bit. Before in the `var.dims()[dim] == 0` case, `cumsum` returned just a copy of `var` which still includes the empty dimension. But with this change, it removes that dim. I think the original approach is better because `cumsum` keeps the summed dim in other cases.

Reply:
Oh, I think you are right, returning `cumulative` is actually wrong, I missed.

Reply:
Revertet